### PR TITLE
Update README.md for how to run examples

### PR DESCRIPTION
**Pull request by nevermosby**
Hi all,

The commit is to provide the helpful guidelines on how to run the examples.

Signed-off-by: Wen-Quan Li <legendarilylwq@gmail.com>

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -4,3 +4,14 @@
 - [uprobe](uprobe/) - Like a kprobe, but for symbols in userspace binaries (e.g. `bash`).
 - [tracepoint](tracepoint/) - Attach a program to predetermined kernel tracepoints.
 - Add your use case(s) here!
+
+## How to run
+
+```bash
+git clone https://github.com/cilium/ebpf.git
+cd examples/[kprobe,uprobe,tracepoint...]
```
Comment (lmb) [inline examples/README.md:12]: You can simplify this to:

```suggestion
cd ebpf/examples/
go run -exec sudo [kprobe,uprobe,tracepoint...]
```

No go build step needed, so you can skip the the bit about `go build main.go`.

**Review by ti-mo — COMMENTED** [review]
Why don't we just add a `make examples` target to the main Makefile instead, and document that here? Not sure if this repo is the place to teach a user about `go build`... :thinking:

**Comment by lmb** [comment]
> Why don't we just add a `make examples` target to the main Makefile instead, and document that here? Not sure if this repo is the place to teach a user about `go build`... thinking

I'd rather not bloat the Makefile more. Users are probably interested in individual examples anyways, how do you run those via the makefile?

Documenting the `go run -exec sudo` trick is worth it IMO.

**Comment by ti-mo** [comment]
> Users are probably interested in individual examples anyways, how do you run those via the makefile?

Oh, I would just suggest `make examples` to build them, not run them. That would leave runnable binaries in the example dirs.

I hadn't used `go run` that way before, I thought it had to be called on a single file. So what about `go run -exec sudo .` in the individual example dirs then instead? That doesn't leave a binary behind in the directory.

`go build .` is functionally equivalent to `go build`, so I would omit the dot in case we go for that.

File: examples/README.md [hunk 2]
```diff
@@ -4,3 +4,14 @@
 - [uprobe](uprobe/) - Like a kprobe, but for symbols in userspace binaries (e.g. `bash`).
 - [tracepoint](tracepoint/) - Attach a program to predetermined kernel tracepoints.
 - Add your use case(s) here!
+
+## How to run
+
+```bash
+git clone https://github.com/cilium/ebpf.git
```
[the remark this answers is not in the document]
Comment (nevermosby) [reply examples/README.md:11]: Agreed, will remove it.

File: examples/README.md [hunk 3]
```diff
@@ -4,3 +4,14 @@
...
+cd examples/[kprobe,uprobe,tracepoint...]
+go build . # You will get the executable file and just run it.
+
+# Do not run `go build main.go` directly for the examples of kprobe and uprobe, otherwise you will get errors like "undefined: KProbeExampleObjects".
```
[the remark this answers is not in the document]
Comment (nevermosby) [reply examples/README.md:15]: It does make sense, I will remove it.

File: examples/README.md (hunk 1 above)
Comment (nevermosby) [reply examples/README.md:12]: I tried "go run -exec sudo kprobe" but got errors: `package kprobe is not in GOROOT (/snap/go/7360/src/kprobe)`. And I tried "go run -exec sudo ./kprobe" and it works. So I will document that instead.

**Comment by nevermosby** [comment]
Just updated based on your suggestions, pls help review.

File: examples/README.md (hunk 1 above)
Comment (ti-mo) [reply examples/README.md]: Thanks!

**Pull request merged**